### PR TITLE
Publish Presences - NiceHash & BetterHash

### PR DESCRIPTION
**Pull request by rhys19**
Hello, I have made some simple presences this is my first time, I have created Nicehash & Betterhash presences.

none of these extensions release PRIVATE information they only say what the page name is your viewing and a small description of it ex: Viewing exchange prices.

I hope this gets published as I think there's a few people out there that would use these :)

Please let me know if there is anything to change on it and I gladly will!

### CODE REVIEW

**Comment by ImRodry** [comment]
Idk how you did that but don’t upload the presence.js file

**Comment by EncryptedDev** [comment]
You also have a few too many `metadata.json`s

**Comment by rhys19** [comment]
oh I didn't know, there was no instructions I could find on submitting pr so just the presence.ts then? as for the manifest do I'm not sure where to put it. do I put in /dist or the main folder

**Comment by ImRodry** [comment]
There are [docs](https://docs.premid.app/en/dev/presence) on how to create a presence and git itself shouldn't ever allow you to publish the .js file, which is why I'm surprised as to how you did that, but please delete multiple files in one commit

**Comment by rhys19** [comment]
Ok. sorry, I deleted them via github not via local files so that's why it deleted it in multiples. so I deleted the manifest.json in the main and kept it in dist and deleted the .js is there anything else?

File: websites/B/BetterHash/presence.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const presence = new Presence({
+    clientId: "916918590354104320"
+  }),
+    strings = presence.getStrings({
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  time = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/B/BetterHash/presence.ts:8]:
```suggestion
  browsingTimestamp = Math.floor(Date.now() / 1000);
```

File: websites/B/BetterHash/presence.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+    largeImageKey: "betterhash_1024x1024",
+    startTimestamp: time
```
Comment (EncryptedDev) [inline websites/B/BetterHash/presence.ts:13]:
```suggestion
    startTimestamp: browsingTimestamp
```

File: websites/N/NiceHash/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "author": {
+    "name": "Rhys19",
+    "id": "675080159736365097"
+  },
+  "service": "NiceHash",
+  "description": {
+    "en": "Nicehash is a crypto currency miner & a hashpower marketplace!"
+  },
+  "url": "www.nicehash.com",
+  "version": "1.0.3",
```
Comment (EncryptedDev) [inline websites/N/NiceHash/dist/metadata.json:12]:
```suggestion
  "version": "1.0.0",
```

File: websites/N/NiceHash/presence.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const presence = new Presence({
+    clientId: "915811466353983519"
+  }),
+    strings = presence.getStrings({
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  time = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/N/NiceHash/presence.ts:8]:
```suggestion
  browsingTImestamp = Math.floor(Date.now() / 1000);
```

File: websites/N/NiceHash/presence.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+    largeImageKey: "nicehash_1024x1024",
+    startTimestamp: time
```
Comment (EncryptedDev) [inline websites/N/NiceHash/presence.ts:13]:
```suggestion
    startTimestamp: browsingTImestamp
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/B/BetterHash/presence.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+const presence = new Presence({
+    clientId: "916918590354104320"
+  }),
+    strings = presence.getStrings({
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  browsingTimestamp = Math.floor(Date.now() / 1000);
+
+  presence.on("UpdateData", async () => {
```
Comment (ImRodry) [inline websites/B/BetterHash/presence.ts:10]: fix indentation (run prettier + eslint)

File: websites/N/NiceHash/presence.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+const presence = new Presence({
+    clientId: "915811466353983519"
+  }),
+    strings = presence.getStrings({
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  browsingTImestamp = Math.floor(Date.now() / 1000);
+
+  presence.on("UpdateData", async () => {
```
Comment (ImRodry) [inline websites/N/NiceHash/presence.ts:10]: ditto

File: websites/N/NiceHash/presence.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    const presenceData: PresenceData = {
+    largeImageKey: "nicehash_1024x1024",
+    startTimestamp: browsingTImestamp
```
Comment (ImRodry) [inline websites/N/NiceHash/presence.ts:13]: that's not how you camelcase
```suggestion
    startTimestamp: browsingTimestamp
```

File: websites/N/NiceHash/presence.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
+const presence = new Presence({
+    clientId: "915811466353983519"
+  }),
+    strings = presence.getStrings({
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  browsingTImestamp = Math.floor(Date.now() / 1000);
```
Comment (ImRodry) [inline websites/N/NiceHash/presence.ts:8]:
```suggestion
  browsingTimestamp = Math.floor(Date.now() / 1000);
```

**Review by ImRodry — CHANGES_REQUESTED** [review]
Please batch these suggestions into one commit and use vscode, not github

**Comment by rhys19** [comment]
> Please batch these suggestions into one commit and use vscode, not github

ok, sorry my internet was out for like 2 hours doing it via vscode now and then uploading with the git client.

**Comment by rhys19** [comment]
I tried using eslint & prettier but in vscode it doesn't do any changes. it had me remove variable that wasn't used but that was it, I'm fairly new to typescript I've only coded javascript and other languages but this is my first for typescript

**Comment by rhys19** [comment]
ESlint & Prettier did do some indentation and spaces I uploaded them via git (Used VSCode).

File: package.json [hunk 10]
```diff
@@ -29,8 +29,8 @@
     "@types/node": "^16.10.2",
     "@types/prettier": "^2.4.1",
     "@types/semver": "^7.3.8",
-    "@typescript-eslint/eslint-plugin": "^4.33.0",
-    "@typescript-eslint/parser": "^4.33.0",
+    "@typescript-eslint/eslint-plugin": "^5.6.0",
+    "@typescript-eslint/parser": "^5.0.0",
```
Comment (Bas950) [inline package.json:33]: undo

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: package.json (hunk 10 above)
Comment (rhys19) [reply package.json]: fixed

**Comment by EncryptedDev** [comment]
Fix this PR and reopen

**Comment by rhys19** [comment]
> Fix this PR and reopen

my question is how come the coinbase extension was merged, when mine isn't that different???

it's my first time doing typescript I don't understand what's wrong with it...
been trying to understand what was wrong but if someone could just point me in the right direction it'd be nice.

**Comment by EncryptedDev** [comment]
Only commit the files you want to change, nothing more. You have commited random files such as: `package.json.bak` and deleted `yarn.lock`

**Pull request closed without merging**